src/clld_markdown_plugin/__init__.py

```diff
@@ -80,40 +99,36 @@ def render_ex(req, objid, table, session, ids=None, **kw):
                 render_ex(req, mid, table, session, subexample=True) for mid in ids[0].split(",")
             ]
             return '\n\n'.join(ex_strs)
-    return rendered_sentence(session.query(Sentence).filter(Sentence.id == objid)[0])
-
-
-def render_cogset(req, objid, table, session, ids=None, **kw):
-    ctx = session.query(UnitParameter).get(objid)
-    return f"""<%util:table items="{ctx.reflexes}" args="item"">
-            <%def name="head()">
-                <th>Morph</th>
-                <th>Language</th>
-                <th>Alignment</th>
-            </%def>
-            <td>${h.link(request, item.counterpart)}</td>
-            <td>${h.link(request, item.counterpart.language)}</td>
-            <td>
-                <span class="alignment">${item.alignment}</span>
-            </td>
-        </%util:table>"""
-
-
-def markdown(req, s, permalink=True, session=None):
+    return rendered_sentence(session.query(common.Sentence).filter(common.Sentence.id == objid)[0])
+
+
+def markdown(req, s: str, permalink=True, session=None) -> str:
+    """
+    :param req:
+    :param s:
+    :param permalink:
+    :param session:
+    :return: `str` containing HTML formatting.
+    """
+    if __name__ not in req.registry.settings:  # pragma: no cover
+        raise KeyError(
+            '{} must be included in the app config to use the "markdown" function.'.format(
+                __name__))
+    settings = req.registry.settings[__name__]
```

Comment:
Reading the plugin settings from the registry circumvents the mutable module global state, which can become hard to maintain. Mutable module-global state basically means using monkey-patching as configuration mechanism, with all the risks in a multi-threaded/multi-process environment.